packages/contents/src/drivefs.ts

```diff
@@ -177,16 +193,16 @@ export class DriveFSEmscriptenStreamOps implements IEmscriptenStreamOps {
     length: number,
     position: number
   ): number {
-    if (length <= 0 || stream.file === undefined) {
+    if (
+      length <= 0 ||
+      stream.file === undefined ||
+      position >= (stream.file.data.length || 0)
```

Comment:
ref
- https://github.com/jupyterlite/jupyterlite/issues/840#issuecomment-1313061817 
- https://github.com/jupyterlite/jupyterlite/commit/c7c29078edf28f1828023385fde8e4d130a4630a#r89808665

There's still _something_ happening in the pyolite case, where doing successive `writes` without closing between them yields a `database disk image is malformed`